core/src/main/java/greencity/controller/TipsAndTricksController.java

```diff
@@ -0,0 +1,149 @@
+     */
+
+    @ApiOperation(value = "Add new tips & tricks.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = HttpStatuses.CREATED,
+            response = AddTipsAndTricksDtoResponse.class),
+        @ApiResponse(code = 303, message = HttpStatuses.SEE_OTHER),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @PostMapping(consumes = {MediaType.APPLICATION_JSON_UTF8_VALUE, MediaType.MULTIPART_FORM_DATA_VALUE})
+    public ResponseEntity<AddTipsAndTricksDtoResponse> save(
+        @ApiParam(value = "Add tips & tricks request", required = true)
+        @RequestPart AddTipsAndTricksDtoRequest addTipsAndTricksDtoRequest,
+        @ApiParam(value = "Tips & tricks image")
+        @RequestPart(required = false) MultipartFile image,
+        @ApiIgnore Principal principal) {
+        return ResponseEntity.status(HttpStatus.CREATED).body(
+            tipsAndTricksService.save(addTipsAndTricksDtoRequest, image, principal.getName()));
+    }
+
+    /**
+     * Method for getting tips & tricks by id.
+     *
+     * @return {@link TipsAndTricksDto} instance.
+     */
+    @ApiOperation(value = "Get tips & tricks by id.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @GetMapping("/{id}")
+    public ResponseEntity<TipsAndTricksDto> getTipsAndTricksById(@PathVariable Long id) {
+        return ResponseEntity.status(HttpStatus.OK).body(tipsAndTricksService.findDtoById(id));
+    }
+
+    /**
+     * Method for getting all tips & tricks by page.
+     *
+     * @return PageableDto of {@link TipsAndTricksDto} instances.
+     */
+    @ApiOperation(value = "Find all tips & tricks by page.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @GetMapping("")
+    @ApiPageable
+    public ResponseEntity<PageableDto<TipsAndTricksDto>> findAll(@ApiIgnore Pageable page) {
+        return ResponseEntity.status(HttpStatus.OK).body(tipsAndTricksService.findAll(page));
+    }
+
+    /**
+     * Method for deleting {@link TipsAndTricks} by its id.
+     *
+     * @param tipsAndTricksId {@link TipsAndTricks} id which will be deleted.
+     * @return id of deleted {@link TipsAndTricks}.
+     */
+    @ApiOperation(value = "Delete tips & tricks.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 303, message = HttpStatuses.SEE_OTHER),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @DeleteMapping("/{tipsAndTricksId}")
```

Comment:
Name variable as id, not {tipsAndTricksId}. You have it in tipsandtricks controller so it understandable that it belongs to this entity

Reply:
Renamed variable to "id"